nvflare/private/fed/server/job_validator.py

```diff
@@ -44,80 +46,125 @@ def validate(self, job_name: str, job_data: bytes) -> Tuple[bool, str, dict]:
         meta = {}
         try:
             with ZipFile(BytesIO(job_data), "r") as zf:
-                meta_file = job_name + "/meta.json"
-                meta_data = zf.read(meta_file)
-                meta = json.loads(meta_data)
-                if JobMetaKey.JOB_FOLDER_NAME not in meta:
-                    meta[JobMetaKey.JOB_FOLDER_NAME.value] = job_name
-
-                self._validate_deployment(meta, zf)
-
-                # Validating resource_spec
-                resource_spec = meta.get(JobMetaKey.RESOURCE_SPEC.value)
-                if resource_spec and not isinstance(resource_spec, dict):
-                    raise ValueError(f"Invalid resource_spec for job {job_name}")
+                meta = self._validate_meta(job_name, zf)
+                site_list = self._validate_deploy_map(job_name, meta, zf)
+                clients = self._get_all_clients(site_list)
+                self._validate_min_clients(job_name, meta, clients)
+                self._validate_resource(job_name, meta)
+                self._validate_mandatory_clients(job_name, meta, clients)
 
         except ValueError as e:
             return False, str(e), meta
 
         return True, "", meta
 
-    def _validate_deployment(self, meta: dict, zip_file: ZipFile) -> None:
+    def _validate_meta(self, job_name: str, zf: ZipFile):
+        meta_file = f"{job_name}/{META}"
+        logging.info(f"validate file {meta_file} exists for job {job_name}")
```

Comment:
DEBUG level maybe good enough for this message.

Reply:
ok, will change